### PR TITLE
made IFilteredTemplateInfo obsolete

### PR DESCRIPTION
**Pull request by vlada-shubina**
### Problem
Part of public API delivery.
made class IFilteredTemplateInfo obsolete - it is duplicate of ITemplateMatchInfo

### Solution
made class IFilteredTemplateInfo obsolete, removed usages

### Checks:
- [ ] Added unit tests
- [x] Added `#nullable enable` to all the modified files [?](https://github.com/dotnet/templating/wiki/Contributing#coding-style)

### CODE REVIEW

File: src/Microsoft.TemplateEngine.Edge/TemplateListFilter.cs [hunk 1]
```diff
@@ -14,7 +16,19 @@ namespace Microsoft.TemplateEngine.Edge
...
     {
-        // Deprecating. ITemplateMatchInfo will eventually fully replace IFilteredTemplateInfo
+        /// <summary>
+        /// Exact match criteria - the templates should match all filters.
+        /// </summary>
+        /// <seealso cref="GetTemplateMatchInfo"/>
+        public static Func<ITemplateMatchInfo, bool> ExactMatchFilter => x => x.IsMatch;
+
+        /// <summary>
+        /// Partial match criteria - the templates should match one of the filters.
```
Comment (DavidKarlas) [inline src/Microsoft.TemplateEngine.Edge/TemplateListFilter.cs:26]: `should match one of the filters` could be `should match at least one of the filters`

File: src/Microsoft.TemplateEngine.IDE/Bootstrapper.cs [hunk 2]
```diff
@@ -59,12 +59,29 @@ public void Register(Assembly assembly)
...
+        /// <summary>
+        /// Gets list of available templates, if <paramref name="filters"/> are provided returns only matching templates.
+        /// </summary>
+        /// <param name="exactMatchesOnly">
+        /// true: templates should match all filters;
+        /// false:  templates should match any filter.
+        /// </param>
+        /// <param name="filters">List of filters to apply. See <see cref="WellKnownSearchFilters"/> for predefined filters.</param>
+        /// <returns>Filtered list of available templates with details on the applied filters matches.</returns>
+        public async Task<IReadOnlyCollection<ITemplateMatchInfo>> GetTemplates(bool exactMatchesOnly = true, params Func<ITemplateInfo, MatchInfo?>[] filters)
```
Comment (DavidKarlas) [inline src/Microsoft.TemplateEngine.IDE/Bootstrapper.cs:78]: add `Async` to end of method name

Comment (DavidKarlas) [reply src/Microsoft.TemplateEngine.IDE/Bootstrapper.cs:78]: probably also want to add `CancellationToken`

**Review by DavidKarlas — CHANGES_REQUESTED** [review]

File: src/Microsoft.TemplateEngine.IDE/Bootstrapper.cs [hunk 3]
```diff
@@ -59,12 +59,29 @@ public void Register(Assembly assembly)
             EnvironmentSettings.SettingsLoader.Components.RegisterMany(assembly.GetTypes());
         }
 
+        [Obsolete("Use GetTemplates instead")]
```
Comment (bozturkMSFT) [inline src/Microsoft.TemplateEngine.IDE/Bootstrapper.cs:62]: Can we use `nameof(GetTemplates)` here to allow auto-refactoring later?

File: src/Microsoft.TemplateEngine.IDE/Bootstrapper.cs [hunk 4]
```diff
@@ -59,12 +59,29 @@ public void Register(Assembly assembly)
...
             EnsureInitialized();
             return TemplateListFilter.FilterTemplates(await EnvironmentSettings.SettingsLoader.GetTemplatesAsync(default).ConfigureAwait(false), exactMatchesOnly, filters);
         }
 
+        /// <summary>
+        /// Gets list of available templates, if <paramref name="filters"/> are provided returns only matching templates.
+        /// </summary>
+        /// <param name="exactMatchesOnly">
+        /// true: templates should match all filters;
+        /// false:  templates should match any filter.
```
Comment (bozturkMSFT) [inline src/Microsoft.TemplateEngine.IDE/Bootstrapper.cs:74]: Number of spaces after semicolon do not match.
Punctuation at the end of the lines do not match.

**Review by bozturkMSFT — APPROVED** [review]

File: src/Microsoft.TemplateEngine.IDE/Bootstrapper.cs (hunk 3 above)
Comment (vlada-shubina) [reply src/Microsoft.TemplateEngine.IDE/Bootstrapper.cs:62]: interpolated strings are not supported in this context

**Review by DavidKarlas — APPROVED** [review]

**Pull request merged**